toolchain/check/testdata/impl/lookup/min_prelude/import.carbon

```diff
@@ -1881,9 +1889,15 @@ fn Test(c: HasExtraInterfaces.C(type)) {
 // CHECK:STDOUT:
 // CHECK:STDOUT: fn @K.2 [from "has_generic_interface.carbon"];
 // CHECK:STDOUT:
-// CHECK:STDOUT: specific @AnyParam(constants.%T, constants.%X) {
+// CHECK:STDOUT: specific @AnyParam(constants.%T, constants.%X.f22ca8.1) {
+// CHECK:STDOUT:   %T => constants.%T
+// CHECK:STDOUT:   %X => constants.%X.f22ca8.1
+// CHECK:STDOUT:   %pattern_type => constants.%pattern_type.7dc
+// CHECK:STDOUT: }
+// CHECK:STDOUT:
+// CHECK:STDOUT: specific @AnyParam(constants.%T, constants.%X.f22ca8.2) {
 // CHECK:STDOUT:   %T => constants.%T
-// CHECK:STDOUT:   %X => constants.%X
+// CHECK:STDOUT:   %X => constants.%X.f22ca8.2
 // CHECK:STDOUT:   %pattern_type => constants.%pattern_type.7dc
 // CHECK:STDOUT: }
```

Comment:
The change here looks potentially incorrect: we have
```
// CHECK:STDOUT:   %X.f22ca8.1: %T = bind_symbolic_name X, 1 [symbolic]
// CHECK:STDOUT:   %X.f22ca8.2: @AnyParam.%T (%T) = bind_symbolic_name X, 1 [symbolic]
```
and specifics with value blocks for both of those. But the second one shouldn't be an unattached constant because its type is attached, and we shouldn't have value blocks for a specific with attached arguments.

Reply:
Fixed, although I'm not especially confident it's the right fix. It looks like almost all the diffs in this PR were a consequence of this issue, even the seemingly-benign ones.

I still need to validate that this hasn't re-broken #5400, but I'm sending it back for more comments in the meantime.

Reply:
Update: I've checked, and this still fixes #5400.